v2/pkg/catalog/disk/path.go

```diff
@@ -49,20 +47,8 @@ var errNoValidCombination = errors.New("no valid combination found")
 
 // tryResolve attempts to load locate the target by iterating across all the folders tree
 func (c *DiskCatalog) tryResolve(fullPath string) (string, error) {
-	dir, filename := filepath.Split(fullPath)
-	pathInfo, err := folderutil.NewPathInfo(dir)
-	if err != nil {
-		return "", err
+	if _, err := os.Stat(fullPath); !os.IsNotExist(err) {
```

Comment:
Aren't we losing the capability to locate the `nuclei-templates` folder on the path tree by removing the iterative logic?

Reply:
This was causing issues where if you mentioned `test.yaml` it was finding any test.yaml in subdirectory even though you wanted one in the current directory. Either we prioritize order better or this seemed like the solution to prevent unintended templates from running.

@Mzack9999 do we have a testcase for this?

Reply:
I see, marking this as resolved